src/app/services/baw-api/baw-api.service.ts

```diff
@@ -281,24 +281,62 @@ export abstract class BawApiService<Model extends AbstractModel> {
    * @param filters Base Filters
    * @param key Foreign key
    * @param model Foreign key value (if undefined, returns base filters)
+   * @param comparison Comparison to be performed
    */
   protected filterByForeignKey(
     filters: Filters<Model>,
-    key: KeysOfType<Model, number | string>,
-    model: AbstractModel | string | number
+    key: KeysOfType<Model, number | string | Set<string> | Set<number>>,
+    model: AbstractModel | string | number,
+    comparison: keyof (Comparisons & Subsets) = "eq"
   ): Filters<Model> {
+    return this.customFilterByForeignKey(filters, key, model, comparison);
+  }
+
+  /**
+   * Modify a base filter to add a foreign key condition
+   *
+   * @param filters Base Filters
+   * @param key Foreign key
+   * @param models Foreign key values (if undefined, returns base filters)
+   * @param comparison Comparison to be performed
+   */
+  protected filterByForeignKeys(
```

Comment:
I don't really like the foreign key term in this case. It is very database specific.

It's more like "filter by association"  right?

`filterThroughAssociation`?